src/pyscipopt/expr.pxi

```diff
@@ -11,21 +55,21 @@ def _is_number(e):
 
 def _expr_richcmp(self, other, op):
     if op == 1: # <=
-        if isinstance(other, Expr):
+        if isinstance(other, Expr) or isinstance(other, GenExpr):
```

Comment:
If `Expr` and `GenExpr` are mostly handled the same, then maybe it would be easier to use a common base class for both of them?

Reply:
yes, maybe, but I felt uncertain. The idea was to try to keep Expr as untouched as possible, given that it has been much more tested.